From ab7540f0e81d1665b57c18c551c63546e0c3a680 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Thu, 29 May 2014 09:58:10 +0800
Subject: [PATCH 01/71] Revert "lsi-dma32: tune the modifier calculation for
 LSI acp"

This reverts commit d2ed0f1cc2298c88cca2fb9df9c92856c531d2c8.

The commit(d2ed0f1cc2) make a fixed length in the SRIO transfer
for acp34xx. But the implement just setup the burst size for 5.
The lsi-axm55xx also need this.

Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 drivers/dma/lsi-dma32.c |   19 +------------------
 1 files changed, 1 insertions(+), 18 deletions(-)

diff --git a/drivers/dma/lsi-dma32.c b/drivers/dma/lsi-dma32.c
index 39a09ca..6386ecd 100644
--- a/drivers/dma/lsi-dma32.c
+++ b/drivers/dma/lsi-dma32.c
@@ -534,23 +534,6 @@ gpdma_prep_memcpy(struct dma_chan *chan,
 	}
 
 	/* Maximize memory access width based on job src, dst and length */
-#ifdef CONFIG_ACP
-	access_size = 5;
-	if (!(size % 16)) {
-		src_size = 16;
-		access_size |= (4 << 3);
-	} else if (!(size % 8)) {
-		src_size = 8;
-		access_size |= (3 << 3);
-	} else if (!(size % 4)) {
-		src_size = 4;
-		access_size |= (2 << 3);
-	} else if (!(size % 2)) {
-		src_size = 2;
-		access_size |= (1 << 3);
-	} else
-		src_size = 1;
-#else
 	switch (ffs((u32)dst | (u32)src | size)) {
 	case 1:
 		src_size = 1;
@@ -573,7 +556,7 @@ gpdma_prep_memcpy(struct dma_chan *chan,
 		access_size = (4 << 3);
 		break;
 	}
-#endif
+
 	ch_dbg(dmac, "dst=%#llx src=%#llx size=%u mod=%d\n",
 		dst, src, size, src_size);
 	x_count = (size/src_size) - 1;
-- 
1.7.5.4

